### PR TITLE
[7.44.x][JBPM-9357] Add springboot Kafka WIH tests

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-9357](https://issues.redhat.com/browse/JBPM-9357)

Refactored integration tests to be included in this repo

**referenced Pull Requests**:  [droolsjbpm-build-bootstrap#1482](https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1482)

### CODE REVIEW

**Comment by gmunozfe** [comment]
@elguardian @afalhambra please review

**Comment by elguardian** [comment]
Jenkins retest this please.

**Review by elguardian — APPROVED** [review]
Waiting for CI, but almost the same code as it was on integrations.
Good to go when CI green

**Comment by gmunozfe** [comment]
jenkins execute cdb

File: jbpm-workitem-itests/pom.xml
Comment (afalhambra) [inline jbpm-workitem-itests/pom.xml:2]:
```suggestion
<!-- ~ Copyright 2020 Red Hat, Inc. and/or its affiliates. ~ ~ Licensed under 
```

File: jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/JBPMApplication.java
Comment (afalhambra) [inline jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/JBPMApplication.java:2]:
```suggestion
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```

File: jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java
Comment (afalhambra) [inline jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java:2]:
```suggestion
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```

File: jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/TestAutoConfiguration.java
Comment (afalhambra) [inline jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/TestAutoConfiguration.java:2]:
```suggestion
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
```

**Comment by mareknovotny** [comment]
Jenkins retest this please

**Review by afalhambra — COMMENTED** [review]
Only copyright year should be updated for some files, but not really an issue, feel free to skip it.
Looks very good to me.

File: jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java
Comment (MarianMacik) [inline jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java:34]: Hmm, so no authentication required to reach endpoints?

File: jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java
Comment (MarianMacik) [inline jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java:181]: Why not to use just sleep here as there is no way to call countDown on the latch anyway.

File: jbpm-workitem-itests/src/test/resources/kjars/kafka-process/src/main/java/com/myspace/kafka_process/BoxSerializer.java
Comment (MarianMacik) [inline jbpm-workitem-itests/src/test/resources/kjars/kafka-process/src/main/java/com/myspace/kafka_process/BoxSerializer.java:29]: I noticed that serializer is in the kjar, but deserializer is outside of it. Is that on purpose?

File: pom.xml
Comment (MarianMacik) [inline pom.xml:1158]: I believe this should be also added to the productized profile. Otherwise we won't be available to run these tests as part of QE certifications. @mareknovotny am I right?

**Review by MarianMacik — COMMENTED** [review]
Looks good, just few ideas/notes.

File: pom.xml
Comment (mareknovotny) [reply pom.xml:1158]: i am not sure how you run tests for binaries ;)

File: jbpm-workitem-itests/pom.xml
Comment (gmunozfe) [reply jbpm-workitem-itests/pom.xml:75]: I'd prefer to keep it as test scope to emphasize that this dependency is just needed for tests (not a strong opinion, it can be also be removed, if you consider it better)

File: jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java
Comment (gmunozfe) [reply jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java:34]: I've kept the same as in springboot module test; in this case, the aim of the tests is not related to authentication

File: jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java
Comment (gmunozfe) [reply jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java:181]: right, this approach is just to avoid sleep and be more sonarcloud rules compliant

File: jbpm-workitem-itests/src/test/resources/kjars/kafka-process/src/main/java/com/myspace/kafka_process/BoxSerializer.java
Comment (gmunozfe) [reply jbpm-workitem-itests/src/test/resources/kjars/kafka-process/src/main/java/com/myspace/kafka_process/BoxSerializer.java:29]: Yes, serializer is needed for publishing the message, meanwhile, deserializer is part of the assertion of the tests (not really needed in the producer side)

File: pom.xml
Comment (gmunozfe) [reply pom.xml:1158]: added, thanks a lot for your review @MarianMacik

Comment (gmunozfe) [reply pom.xml:1158]: I think if it's productized it will affect also to the total number of modules, than @jstastny-cz modified also here:
https://github.com/kiegroup/jbpm-work-items/pull/156
So, when that PR is merged, I'd increase also here

File: jbpm-workitem-itests/pom.xml
Comment (gmunozfe) [reply jbpm-workitem-itests/pom.xml]: done

File: jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/JBPMApplication.java
Comment (gmunozfe) [reply jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/JBPMApplication.java]: done

File: jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java
Comment (gmunozfe) [reply jbpm-workitem-itests/src/main/java/org/jbpm/workitem/springboot/samples/WebSecurityConfig.java]: done

File: jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/TestAutoConfiguration.java
Comment (gmunozfe) [reply jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/TestAutoConfiguration.java]: done

File: pom.xml
Comment (MarianMacik) [reply pom.xml:1158]: @mareknovotny I rather meant that these are the only modules which are built by productization. So if we don't include it there (in the productized profile) we won't see this module and its tests in the product -> no testing.
@gmunozfe thanks for increase as well

File: repository/pom.xml
Comment (MarianMacik) [inline repository/pom.xml:682]: Should it be 8? Not just 7?

File: jbpm-workitem-itests/src/test/resources/kjars/kafka-process/src/main/java/com/myspace/kafka_process/BoxSerializer.java
Comment (MarianMacik) [reply jbpm-workitem-itests/src/test/resources/kjars/kafka-process/src/main/java/com/myspace/kafka_process/BoxSerializer.java:29]: I see, thanks.

File: jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java
Comment (MarianMacik) [reply jbpm-workitem-itests/src/test/java/org/jbpm/workitem/springboot/samples/KafkaProxySampleTest.java:181]: OK, makes sense now.

File: repository/pom.xml
Comment (gmunozfe) [reply repository/pom.xml:706]: kafka-item was also missing (check #156 )

Comment (gmunozfe) [reply repository/pom.xml:706]: @mareknovotny apart from -Pproductized, what else is needed to execute repository tests?

Comment (jstastny-cz) [reply repository/pom.xml:706]: I think module number increase is understandable (we've added the itest module here), I just don't understand the increase in total.wi.count. Tests locally are failing for me with:
```
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[WARNING] The expression ${artifactId} is deprecated. Please use ${project.artifactId} instead. @ 
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead. @ 
[ERROR] 'dependencies.dependency.version' for org.testcontainers:kafka:jar is missing. @ line 130, column 17
[ERROR] 'dependencies.dependency.version' for org.testcontainers:toxiproxy:jar is missing. @ line 135, column 17
```
Is it possible that there needs to be done something around dependency management after the move to this repo?

Comment (gmunozfe) [reply repository/pom.xml:706]: @jstastny-cz it's referenced in the first comment: [droolsjbpm-build-bootstrap#1482](https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1482)

Comment (gmunozfe) [reply repository/pom.xml:706]: @jstastny-cz kept wi.count in 16, thanks!

**Comment by mareknovotny** [comment]
Is it ready to be merged?

**Review by MarianMacik — APPROVED** [review]

**Pull request merged**
